### DISCUSSION TITLE
Bad Installation? Cant start streamlink or watch a Stream. 404/410 Client Error.

### DISCUSSION THREAD

**Discussion by 3Sawyer** [streamlink/streamlink#4059] [Q&A] [answered]
Checklist
- [x] I understand the difference between Streamlink and Streamlink Twitch GUI.
- [x] I tried solving my issue by reading the project's wiki first.
- [x] This is not a bug report or feature request and I have read the contribution guides.

Environment details
Operating system and version: 
Zorin OS 15.3 based on Ubuntu 18.04 Bionic
Intel® Core™2 Duo CPU P8600 @ 2.40GHz × 2 
64-Bit

Streamlink Twitch GUI version:
streamlink-twitch-gui-v1.13.0-x86_64
Streamlink version:
Tried to install streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64 but im not sure
Configuration details:

Additional comments:

Im really confused. 
I tried to install steamlink with Terminal and later with AppImage. 
I dont know if i have several steamlink Installations. Im really new to Linux. 

First i downloaded streamlink-twitch-gui-v1.13.0-linux64.tar.gz. Extracted the Folder and run 
`'/home/ryda/Downloads/streamlink-twitch-gui-v1.13.0-linux64/streamlink-twitch-gui/add-menuitem.sh' `
But it didnt work so i removed it again with
`'/home/ryda/Downloads/streamlink-twitch-gui-v1.13.0-linux64/streamlink-twitch-gui/add-menuitem.sh' `
I then tried the AppImage Version. I installed AppImage and installed `streamlink-twitch-gui-v1.13.0-x86_64`
But i cant open a stream with the Terminal. Streamlink doesnt start. So i tried to install Streamlink AppImage Version 
streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage
But that doesnt work. It doesnt open in AppImage.
Im very very confused what to do now. 
I paste the Error Messages i get in the Terminal.

This happens when i try to open a Youtube Video
>~$ streamlink https://www.youtube.com/watch?v=iTFtY6KgS58 best
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=iTFtY6KgS58
error: Unable to open URL: https://youtube.com/get_video_info (410 Client Error: Gone for url: https://www.youtube.com/get_video_info?video_id=iTFtY6KgS58&el=detailpage)

This happens when i try to open a Twitch Stream
> ~$ streamlink twitch.tv/esl_csgo 720p60
[cli][info] Found matching plugin twitch for URL twitch.tv/esl_csgo
error: Unable to open URL: https://tmi.twitch.tv/hosts (404 Client Error: Not Found for url: https://tmi.twitch.tv/hosts?as3=t&include_logins=1&host=31239503)

I dont know what --stdout | vlc - means. I found it somewhere and with this it will open VLC but i get an Error in the VLC Player:
`Your input can't be opened:
VLC is unable to open the MRL 'fd://0'. Check the log for details.`

And Terminal Error:
>~$ streamlink twitch.tv/esl_csgo 720p --stdout | vlc -
[cli][info] Found matching plugin twitch for URL twitch.tv/esl_csgo
error: Unable to open URL: https://tmi.twitch.tv/hosts (404 Client Error: Not Found for url: https://tmi.twitch.tv/hosts?as3=t&include_logins=1&host=31239503)
VLC media player 3.0.16 Vetinari (revision 3.0.16-0-g5e70837d8d)
[...] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
Qt: Session management error: Could not open network socket
QApplication: invalid style override 'adwaita' passed, ignoring it.
	Available styles: Breeze, Windows, Fusion
[...] mjpeg demux error: cannot peek
QObject::~QObject: Timers cannot be stopped from another thread

I tried installing the StreamLink AppImage with
`~$ chmod +x streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage
chmod: cannot access 'streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage': No such file or directory
`
And when i try `~$ cd ~/Downloads/`
`~$ cd ~/Downloads/
~/Downloads$ chmod +x streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage
~/Downloads$ ./streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage`
this happens:

> Gtk-Message: 01:21:31.876: Failed to load module "canberra-gtk-module"
Gtk-Message: 01:21:31.879: Failed to load module "canberra-gtk-module"
QApplication: invalid style override passed, ignoring it.
ERROR: ld.so: object '/usr/lib/appimagelauncher/libbinfmt-bypass-preload.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
execv error: No such file or directory
` sudo apt-get install libcanberra-gtk*
`
Didnt help either...Same Error...

 I really dont know what to do. 
I tried:
`sudo apt install streamlink
[sudo] password for ryda:      
Reading package lists... Done
Building dependency tree       
Reading state information... Done
streamlink is already the newest version (0.10.0+dfsg-1).
The following packages were automatically installed and are no longer required:
  linux-hwe-5.4-headers-5.4.0-77 linux-hwe-5.4-headers-5.4.0-80
  linux-hwe-5.4-headers-5.4.0-81 linux-hwe-5.4-headers-5.4.0-84
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
`
Doesnt change anything. Same Error Messages. 
I would be very happy for easy to follow Tips as i mentioned im very new to Linux and hope somebody can help me use this great Software. 

€dit: This Terminal Error is only visible for 1 second when i try to start streamlink from the Main Menu.
![image](https://user-images.githubusercontent.com/83242863/135538321-3e25bfad-107c-4a19-b920-8e57fe52e1ef.png)
I installed it with 
`sudo apt install libcanberra-gtk-module libcanberra-gtk3-module`
But there is still the same Error only visible for 1 second.

**Comment by bastimeyer** [comment, marked as answer]
You are doing a lot of things wrong. I'm not sure if it makes sense responding to all of them and just tell you what to do right.

First off, you are on ZorinOS, which is based on Ubuntu. Streamlink's Ubuntu packages are out of date and if you install Streamlink on Ubuntu or Ubuntu-based distros via the default package repositories, then you will have an outdated version installed. You can check this via `streamlink --version` or `streamlink --loglevel debug` and then compare it with the latest version in streamlink's docs or on the github releases page.

This is where the 4xx errors are coming from, because the old versions are incompatible with the current state of these websites you're trying to access and you will need the latest version of Streamlink.

That's why you can either set up a custom package repository on Ubuntu (described in the documentation) to get more recent packages of streamlink on older distro releases or you can download the latest appimage, which is a standalone application as a single executable file and the recommended way on these distros.

If you run `streamlink` in your command line shell, then your shell will look for the `streamlink` executable in the paths defined by the `PATH` environment variable (`echo $PATH` / `which streamlink`). This will be the executable from the old and outdated install, which you should uninstall.

The appimages need to be made executable first before you can run them, as described in the documentation. This can be done via the `chmod` utility, which takes the `+x` parameter ("make executable") and the path to the file you want to make executable. The path depends on your current directory if you've set a relative path. If you don't have any experience with the command line shell, then you can also use your system's GUI file browser to make the file executable. Just right click the file and open the file properties window.

After you've done that, you'll have to execute it like this:
instead of `streamlink STREAMLINK ARGUMENTS HERE`
you run `./streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage STREAMLINK ARGUMENTS HERE`.
Notice the leading `./`, which means "from this directory" instead of "search globally for an executable named like this". If the file is not in the current directory, you can of course also choose a relative or an absolute path.

> wrong ELF class: ELFCLASS32

This means that you have chosen the wrong appimage file. You'll have to get the right one for your CPU and system architecture. Run `uname -m` to see which one it is.

> Streamlink Twitch GUI

If you're trying to use Streamlink Twitch GUI, then this all also applies to its own appimage. Download the right appimage file and make it executable.

Then in its streaming settings menu you'll have to select the streamlink appimage file as streamlink executable, because otherwise it'll look for it on your system globally.

> libcanberra-gtk

This is completely irrelevant, and is just an error message from the Chromium component of Streamlink Twitch GUI which can be ignored. After you've made the appimage of the Twitch GUI executable, you can just run it from your desktop environment / launcher instead of the command line shell, as it doesn't require any command line parameters like streamlink.

**Reply by 3Sawyer** [reply]
Thank you for your reply. I think the easiest way for me is to use AppImage.
The strange thing is it worked with the streamlink-twitch-gui.AppImage (AppImage created a Folder Home/Applications so i guess that worked) but i cant get the streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage to work.
When i use `uname -m` i get `x86_64` so i think "streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage" is the correct Version.

Maybe i need to remove the old streamlink outdated Version first before?
What is the best way of doing that?
`$ sudo apt-get remove --auto-remove streamlink`?
or `$ sudo apt-get purge --auto-remove streamlink`?

**Reply by bastimeyer** [reply]
Post the output of both commands:
```sh
ls -l ./streamlink*
./streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage --loglevel debug
```
when you're in the directory where you've downloaded the appimages.

You'll have to uninstall the streamlink system package, because it's outdated, as I already said. After you've uninstalled it, there won't be the regular `streamlink` command available, because that's part of the package (`/usr/bin/streamlink`).

**Reply by 3Sawyer** [reply]
```
~/Downloads$ ls -l ./streamlink*
-rwxrwxr-x 1 ryda ryda  29523944 Okt  1 01:12 ./streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage
-rw-rw-r-- 1 ryda ryda 116602222 Sep 30 21:56 ./streamlink-twitch-gui-v1.13.0-linux64.tar.gz
./streamlink-twitch-gui-v1.13.0-linux64:
total 4
drwxrwxr-x 7 ryda ryda 4096 Sep 30 21:57 streamlink-twitch-gui
```

```
~/Downloads$ ./streamlink-2.4.0-1-cp39-cp39-manylinux2014_x86_64.AppImage --loglevel debug
Gtk-Message: 02:52:18.057: Failed to load module "canberra-gtk-module"
Gtk-Message: 02:52:18.060: Failed to load module "canberra-gtk-module"
QApplication: invalid style override passed, ignoring it.
ERROR: ld.so: object '/usr/lib/appimagelauncher/libbinfmt-bypass-preload.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
execv error: No such file or directory
```
Sorry im very new with what command should i uninstall the streamlink system package?

**Reply by bastimeyer** [reply]
> what command should i uninstall the streamlink system package?

If you're unsure how to use your package manager, google it or run `man apt` / `man apt-get`.
If `which streamlink` doesn't yield anything, then you've uninstalled it.

> ERROR: ld.so: object '/usr/lib/appimagelauncher/libbinfmt-bypass-preload.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.

How did you install appimagelauncher? This is causing your problem. Remove it. You don't need it.

**Reply by 3Sawyer** [reply]
> How did you install appimagelauncher? This is causing your problem. Remove it. You don't need it.

When i remember correctly. I downloaded "appimagelauncher_2.2.0-travis995.0f91801.bionic_i386.deb"
And installed it with my Software Center. 

I uninstalled streamlink now. `which streamlink` doesnt yield anything. 
And i uninstalled the appimagelauncher too. `which appimagelauncher` yields nothing. 
And i removed  `$ sudo apt-get purge --auto-remove libcanberra-gtk*`
I hope this was also correct because i dont need it?

**Reply by bastimeyer** [reply]
> i386

This is a x86 ("32 bit") package, which is incompatible with your system.

Appimagelauncher installs hooks on your system for intercepting the execution of appimages, so that it can do its appimage management. It's possible that you'll have to restart your system after uninstalling the package to be completely sure.

Appimagelauncher is optional and just a utility for managing appimages. You should be able to launch the streamlink and streamlink twitch gui appimages regularly now. Can you confirm that?

Remember that you'll have to update the streamlink executable in the twitch GUI's streaming settings menu to the streamlink appimage file, so that it uses it for launching streams.

**Reply by 3Sawyer** [reply]
Im very thankful for your help and dont want to ask stupid questions but was it okay to `$ sudo apt-get purge --auto-remove libcanberra-gtk*` ?
Im afraid that i made a mistake and when i reboot everything is broken. I dont really know the difference between `libcanberra-gtk3-module` and `libcanberra-gtk*`
When you say its okay i reboot and try to launch the streamlink appimages.

**Reply by bastimeyer** [reply]
If you've installed an additional package previously and then removed it, there shouldn't be any issues, but that depends on whether you've also removed other packages, which I can't tell you. Your package manager asks you about this though, and it logs everything as well, so if you're unsure, google it.

**Reply by 3Sawyer** [reply]
Im fucked 😅
I only have a Terminal after reboot. And even when i reinstall libcanberra-gtk* i cant see my Desktop. Any ideas what to do?
![1633054481976-692443429](https://user-images.githubusercontent.com/83242863/135555260-1a7deb59-e4ea-4eef-9750-10bebcb271a8.jpg)

**Reply by gravyboat** [reply]
> Any ideas what to do?

You're going to have to ask for help elsewhere or troubleshoot on your own. @bastimeyer has gone way above and beyond in terms of providing support far outside the scope of Streamlink. We cannot teach people how to use their operating system. You may want to see if local classes are available, review documentation, or watch videos on the subject.
